src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,101 @@
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+public class ShooterSubsystem extends SubsystemBase {
+    private static final TalonFX top = new TalonFX(ShooterConstants.TOP_SHOOTER_MOTOR_ID, "canivoreBus");
+    private static final TalonFX bottom = new TalonFX(ShooterConstants.BOTTOM_SHOOTER_MOTOR_ID, "canivoreBus");
+
+    //enum that contains all possible Shooter Speed states for the motors
+    public enum ShooterSpeed {
+        SHUTTLE,
+        SPEAKER,
+        STOPPED,
+        //SPEAKER, //Vision Subsystem?
+        AMP,
+        TRAP,
+        BUMP_FIRE,
+        DUMP;
+    }
+
+    private ShooterSpeed currentState = ShooterSpeed.STOPPED;
+
+    public ShooterSubsystem() {
+        TalonFXConfigurator topConfigurator = top.getConfigurator();
+        TalonFXConfigurator bottomConfigurator = bottom.getConfigurator();
+        TalonFXConfiguration topConfiguration = new TalonFXConfiguration();
+        TalonFXConfiguration bottomConfiguration = new TalonFXConfiguration();
+
+        topConfiguration.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+        topConfiguration.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+        topConfiguration.FutureProofConfigs = true;
+
+        bottomConfiguration.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+        bottomConfiguration.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+        bottomConfiguration.FutureProofConfigs = true;
+
+        top.getRotorVelocity().waitForUpdate(0.05);
+        top.getRotorPosition().waitForUpdate(0.01);
+        topConfigurator.apply(topConfiguration);
+
+        bottom.getRotorVelocity().waitForUpdate(0.05);
+        bottom.getRotorPosition().waitForUpdate(0.01);
+        bottomConfigurator.apply(bottomConfiguration);
+    }
+    private void setShooterState(ShooterSpeed shooterState){
+        switch(shooterState) {
+            case STOPPED -> {
+                top.set(ShooterConstants.STOP_SPEED);
+                bottom.set(ShooterConstants.STOP_SPEED);
+            }
+            case AMP -> {
+                top.set(ShooterConstants.TOP_AMP_SPEED);
+                bottom.set(ShooterConstants.BOTTOM_AMP_SPEED);
+            }
+            case TRAP -> {
+                top.set(ShooterConstants.TOP_TRAP_SPEED);
+                bottom.set(ShooterConstants.BOTTOM_TRAP_SPEED);
+            }
+            case BUMP_FIRE -> {
+                top.set(ShooterConstants.TOP_BUMP_SPEED);
+                bottom.set(ShooterConstants.BOTTOM_BUMP_SPEED);
+            }
+            case DUMP -> {
+                top.set(ShooterConstants.TOP_DUMP_SPEED);
+                bottom.set(ShooterConstants.BOTTOM_DUMP_SPEED);
+            }
+            case SHUTTLE -> {
+                //TODO: Vision calculations for shuttle shot
+            }
+            case SPEAKER -> {
+                //TODO: Vision calculations for speaker shot
+            }
+        }
```

Comment:
this isn't quite right. the state of the subsystem is never being read, so there's no real difference here with just setting the motor speeds directly.you should have methods that set the state, and a switch statement in the periodic method that is constantly checking the state and setting motor speeds directly accordingly

Reply:
no, every subsystem has a periodic method that runs the same way robotPeriodic runs. that's where the switch statement should go

Reply:
ah i see